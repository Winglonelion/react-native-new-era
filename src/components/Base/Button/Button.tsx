import TitleText from 'components/Base/Text/TitleText';
import React, { memo } from 'react';
import {
  ActivityIndicator,
  ColorValue,
  GestureResponderEvent,
  Pressable,
  StyleSheet,
  ViewStyle,
} from 'react-native';

import Colors from 'theme/colors';

interface PropTypes {
  loading?: boolean;
  disabled?: boolean;
  text?: string;
  hideContentWhileLoading?: boolean;
  buttonStyle?: ViewStyle;
  tintColor?: ColorValue;
  round?: boolean;
  onPress?: ((event: GestureResponderEvent) => void) | null;
}

const Button: React.FC<PropTypes> = memo(
  ({
    text,
    loading,
    hideContentWhileLoading,
    disabled,
    tintColor = Colors.black,
    round,
    buttonStyle,
    onPress,
  }) => {
    const isHideContent = loading && hideContentWhileLoading;
    return (
      <Pressable
        onPress={onPress}
        disabled={disabled || loading}
        style={[
          styles.buttonView,
          round && styles.buttonRoundView,
          { backgroundColor: tintColor },
          buttonStyle,
        ]}>
        {text && !isHideContent && (
          <TitleText style={styles.buttonText}>{text}</TitleText>
        )}
        {loading && <ActivityIndicator color={Colors.white} />}
      </Pressable>
    );
  },
);

const styles = StyleSheet.create({
  buttonView: {
    width: 168,
    height: 42,
    borderRadius: 42,
    justifyContent: 'center',
    alignItems: 'center',
  },
  buttonRoundView: {
    width: 80,
    height: 80,
    borderRadius: 40,
  },
  buttonText: {
    fontWeight: '700',
    color: Colors.white,
  },
});

export default Button;
