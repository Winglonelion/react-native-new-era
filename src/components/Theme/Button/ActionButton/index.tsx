export { default as default } from './ActionButton';
