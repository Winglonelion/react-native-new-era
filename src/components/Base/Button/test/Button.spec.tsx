import React from 'react';
import { shallow } from 'enzyme';
import Button, { PropTypes } from '../Button';

const setup = (propOverrides: PropTypes = {} as PropTypes) => {
  const props = { ...propOverrides };
  const wrapper = shallow(<Button {...props} />);
  return {
    props,
    wrapper,
  };
};

describe('Button check snapshot', () => {
  test('should match snapshot', () => {
    const { wrapper } = setup();
    expect(wrapper).toMatchSnapshot();
  });
});
describe('Button check render correctly', () => {
  test('should contain 1 Pressable', () => {
    const { wrapper } = setup();
    expect(wrapper.find('Pressable')).toHaveLength(1);
  });
});
