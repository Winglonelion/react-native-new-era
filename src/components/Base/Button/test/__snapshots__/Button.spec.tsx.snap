// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button check snapshot should match snapshot 1`] = `
<Pressable
  style={
    Array [
      Object {
        "alignItems": "center",
        "borderRadius": 42,
        "height": 42,
        "justifyContent": "center",
        "width": 168,
      },
      undefined,
      Object {
        "backgroundColor": "#000000",
      },
      undefined,
    ]
  }
/>
`;
